python/CMakeLists.txt

```diff
@@ -152,72 +152,5 @@ endif()
 set_property(TARGET CUDAQuantumPythonModules.extension._quakeDialects.dso 
              APPEND PROPERTY INSTALL_RPATH "${_origin_prefix}:${_origin_prefix}/../../lib/")
 
-
-## TODO: Clean-up following from 'old' CMakeLists.txt
-## FIXME: Delete '_pycudaq' target?
-set(LIBRARY_NAME _pycudaq)
-if (NOT SKBUILD)
-  SET(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_RPATH}:$ORIGIN/../../lib:$ORIGIN/../../lib/plugins")
-else()
-  SET(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_RPATH}:$ORIGIN/../cuda_quantum.libs")
-endif()
-pybind11_add_module(_pycudaq 
-                     runtime/cudaq/target/py_runtime_target.cpp 
-                     runtime/cudaq/target/py_testing_utils.cpp
-                     runtime/cudaq/builder/py_QuakeValue.cpp
-                     runtime/cudaq/algorithms/py_observe.cpp
-                     runtime/cudaq/algorithms/py_sample.cpp
-                     runtime/cudaq/algorithms/py_optimizer.cpp
-                     runtime/cudaq/algorithms/py_state.cpp
-                     runtime/cudaq/algorithms/py_vqe.cpp
-                     runtime/cudaq/spin/py_matrix.cpp
-                     runtime/cudaq/spin/py_spin_op.cpp
-                     runtime/cudaq/kernels/py_chemistry.cpp
-                     runtime/common/py_ObserveResult.cpp
-                     runtime/common/py_SampleResult.cpp                     
-                     runtime/common/py_NoiseModel.cpp)
-
-target_compile_options(_pycudaq PRIVATE -Wno-attributes)                     
-target_include_directories(_pycudaq PRIVATE .)
-target_link_libraries(_pycudaq 
-    PRIVATE 
-      nvqir
-      cudaq 
-      cudaq-builder 
-      cudaq-em-default 
-      cudaq-platform-default 
-      cudaq-py-utils)
-
-if (NOT SKBUILD)
-  install(DIRECTORY cudaq DESTINATION .)
-  install(TARGETS _pycudaq DESTINATION cudaq)
-  # Also move these to the build directory for testing.
-  file (COPY cudaq DESTINATION ${CMAKE_BINARY_DIR}/python/)
-  add_custom_command(TARGET _pycudaq POST_BUILD 
-      COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_FILE:_pycudaq> 
-              ${CMAKE_BINARY_DIR}/python/cudaq)
-else()
-  install(TARGETS _pycudaq LIBRARY DESTINATION cudaq)
-endif()
-
 add_subdirectory(runtime/cudaq/distributed)
 add_subdirectory(runtime/cudaq/domains/plugins)
-
-# Do we have pytest and numpy? 
-execute_process(COMMAND ${Python_EXECUTABLE} -m pytest --version
-  OUTPUT_VARIABLE PYTEST_output
-  ERROR_VARIABLE  PYTEST_error
-  RESULT_VARIABLE PYTEST_result)
-execute_process(COMMAND ${Python_EXECUTABLE} -c "import numpy"
-  OUTPUT_VARIABLE NUMPY_output
-  ERROR_VARIABLE NUMPY_error
-  RESULT_VARIABLE NUMPY_result)
-
-if(CUDAQ_BUILD_TESTS)
-  if(${PYTEST_result} EQUAL 0 AND ${NUMPY_result} EQUAL 0)
-    message(STATUS "Pytest and Numpy found, building Python tests.")
-    add_subdirectory(tests)
```

Comment:
`add_subdirectory(tests)` - Don't we need this for testing?

Reply:
I was wondering that - it seems like the CI testing always directly calls `python3 -m pytest ....` so there is no need for the CTest targets, maybe I'm wrong?

Reply:
I see that we use the `CUDAQ_BUILD_TESTS` build argument to skip this target - which is used when Python wheel is built. Additionally, perhaps anyone directly building from source may find it useful (non-CI use case).

Reply:
Sounds good, I'll put it back